src/core/eme/create_session.ts

```diff
@@ -90,6 +92,62 @@ function loadPersistentSession(
     }).do(() => $storedSessions.add(initData, session));
 }
 
+/**
+ * Handle cases where session is persistent.
+ *
+ * If a persisted session exists in the store associated to this initData,
+ * we reuse it without a new license request through the `load` method.
+ * If we load a unusable session, create a new session from scratch.
+ *
+ * @param {Uint8Array} initData
+ * @param {string} initDataType
+ * @param {MediaKeySession|Object} session
+ * @param {MediaKeys|Object} mediaKeys
+ */
+function handlePersistentSession(
+  initData: Uint8Array,
+  initDataType: string,
+  session: IMediaKeySession|MediaKeySession,
+  mediaKeys: MediaKeys|IMockMediaKeys
+): Observable<IGetSessionEvent> {
+
+    let persistentSession: MediaKeySession|IMediaKeySession = session;
+    const storedEntry = $storedSessions.get(initData);
+
+    if (storedEntry) {
+      return loadPersistentSession(
+        storedEntry.sessionId, initData, initDataType, session
+      ).map((evt) => {
+        if (
+          evt.type === "loaded-persistent-session" &&
+          !isLoadedSessionUsable(evt.value.session)
+        ) {
+          // Recreate a new session from scratch.
+          $loadedSessions.closeSession(evt.value.session);
+          $storedSessions.delete(new Uint8Array(initData));
+
+          persistentSession =
+            (mediaKeys as any).createSession("persistent-license"); // TS Bug
+
+          $loadedSessions.add(initData, initDataType, session);
```

Comment:
isn't that already done? We must be careful now, adding the same initData multiple times throw now

Reply:
How, maybe you meant ``persistentSession`` here.